app/scraping.py

```diff
@@ -1,19 +1,47 @@
 from bs4 import BeautifulSoup
 import os
+import requests
+from PIL import Image
+from io import BytesIO
+import logging
 
+# Define global variables for minimum image dimensions
+MIN_IMAGE_WIDTH = 100
+MIN_IMAGE_HEIGHT = 100
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
 
 def extract_images(soup: BeautifulSoup) -> list:
     images = []
     n_images = int(os.getenv("N_IMAGES", 4))  # Default to 4 if not set
+
     for img in soup.find_all(["img", "meta", "link", "source"]):
+        img_url = None
         if img.name == "img" and img.get("src"):
-            images.append(img["src"])
+            img_url = img["src"]
         elif img.name == "meta" and img.get("content") and "image" in img.get("property", ""):
-            images.append(img["content"])
+            img_url = img["content"]
         elif img.name == "link" and img.get("href") and "image" in img.get("type", ""):
-            images.append(img["href"])
+            img_url = img["href"]
         elif img.name == "source" and img.get("srcset"):
-            images.append(img["srcset"].split()[0])  # Take the first URL in srcset
+            img_url = img["srcset"].split()[0]  # Take the first URL in srcset
+
+        if img_url:
+            try:
+                response = requests.get(img_url)
```

Comment:
Seeing ERROR:app.scraping:RequestException for URL https://www.tob-dev.com/wp-content/uploads/2017/11/travel-to-bermuda.png: 406 Client Error: Not Acceptable for url: https://www.tob-dev.com/wp-content/uploads/2017/11/travel-to-bermuda.png

Reply:
The 406 Client Error indicates that the server is not able to provide the requested content. This could be due to the server's configuration or the headers being sent with the request. I will update the code to include appropriate headers to avoid this issue. Thank you for pointing it out!